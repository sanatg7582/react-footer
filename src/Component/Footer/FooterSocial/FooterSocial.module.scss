.FooterSocialLogo {
  background-color: #666666;
  color: #fff;
  position: relative;
  top: 230px;
  height: 90px;
  .FooterSocial {
    width: 50%;
    float: left;
    text-align: left;
    padding: 15px;
    .FooterSocialIcon {
      display: flex;
    }
    span {
      margin: 6px 6px 0px 6px;
    }
  }
  .FooterLogo {
    width: 50%;
    float: left;
  }
}
