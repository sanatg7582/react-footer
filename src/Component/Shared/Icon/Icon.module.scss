.iconImage {
  padding: 5px;
}
