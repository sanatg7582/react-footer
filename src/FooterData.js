const footerData = [
  {
    headingId: 1,
    heading: "Customer Service",
    Items: [
      { ItemId: 1, title: "Help Center" },
      { ItemId: 2, title: "Contact Us" },
      { ItemId: 3, title: "Return an Item" },
      { ItemId: 4, title: "Recall Information" },
      { ItemId: 5, title: "Safety Data Sheets" }
    ]
  },
  {
    headingId: 2,
    heading: "Corporate Info",
    Items: [
      { ItemId: 1, title: "Stapies Business Advantage Overview" },
      { ItemId: 2, title: "Industries Served" },
      { ItemId: 4, title: "Blog" },
      { ItemId: 5, title: "Our Brand" }
    ]
  },
  {
    headingId: 3,
    heading: "Staples Corporate Solution",
    Items: [
      { ItemId: 1, title: "Office Products" },
      { ItemId: 2, title: "Technology Products" },
      { ItemId: 3, title: "Furniture" },
      { ItemId: 4, title: "Facilites" },
      { ItemId: 5, title: "Promotional Products" }
    ]
  },
  {
    headingId: 4,
    heading: "New Customer",
    Items: [
      { ItemId: 1, title: "Become a Customer" },
      { ItemId: 2, title: "Govt. Customers" },
      { ItemId: 3, title: "Memberships" }
    ]
  },
  {
    headingId: 5,
    heading: "Join the Staples Team",
    Items: [{ ItemId: 1, title: "Careers" }]
  }
];
const footerSocialIcons = [
  { IconId: 1, title: "Linkdin", file: "linkedin.png" },
  { IconId: 2, title: "Twitter", file: "twitter.png" },
  { IconId: 3, title: "Facebook", file: "facebook.png" },
  { IconId: 4, title: "Youtube", file: "youtube.png" },
  { IconId: 5, title: "Instagram", file: "insta.png" }
];
const IMAGE_PATH = "assets/images/icons/";
export { footerData, footerSocialIcons, IMAGE_PATH };
