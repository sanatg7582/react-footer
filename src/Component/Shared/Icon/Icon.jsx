import React from "react";
import PropTypes from "prop-types";
import styles from "./Icon.module.scss";
import { IMAGE_PATH } from "../../../FooterData";
/**
 * Icon function return Icon anywhare where you want
 * @param {string} fileName img of src isrequired
 * @param {string} altTitle img of alt isrequired
 * return img element
 */
const Icon = props => {
  const { fileName, altTitle } = props;
  return (
    <img
      className={styles.iconImage}
      src={`${IMAGE_PATH}${fileName}`}
      alt={altTitle}
    />
  );
};
// propsType validation here
Icon.propTypes = {
  fileName: PropTypes.string.isRequired,
  altTitle: PropTypes.string.isRequired
};
export default Icon;
