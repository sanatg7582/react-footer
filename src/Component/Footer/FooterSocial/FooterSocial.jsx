import React from "react";
import PropTypes from "prop-types";
import Icon from "../../Shared/Icon";
import styles from "./FooterSocial.module.scss";
/**
 * FooterSocial function for social icon which we are showing on footer section
 */
const FooterSocial = props => {
  const { footerSocialIcons } = props;

  return (
    <div className={styles.FooterSocialLogo}>
      <div className={styles.FooterSocial}>
        <span>Join us on</span>
        <div className={styles.FooterSocialIcon}>
          {footerSocialIcons.map(value => {
            return (
              <Icon
                key={value.IconId}
                altTitle={value.title}
                fileName={value.file}
              />
            );
          })}
        </div>
      </div>
      <div className={styles.FooterLogo}></div>
    </div>
  );
};
// default props value
FooterSocial.defaultProps = {
  footerSocialIcons: []
};
// propTypes validation
FooterSocial.propTypes = {
  footerSocialIcons: PropTypes.array
};
export default FooterSocial;
