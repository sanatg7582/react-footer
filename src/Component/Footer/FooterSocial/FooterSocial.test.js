import React from "react";
import { mount } from "enzyme";
import FooterSocial from "./FooterSocial";
import Icon from "../../Shared/Icon";
import { footerSocialIcons, IMAGE_PATH } from "../../../FooterData";
describe("renders FooterSocial without crashing ", () => {
  const wrapper = mount(<FooterSocial footerSocialIcons={footerSocialIcons} />);

  it("Load FooterSocal without crash", () => {
    expect(wrapper).toBeTruthy();
  });

  it("footer icon text should be same", () => {
    expect(wrapper.find("FooterSocial").text("span")).toEqual("Join us on");
  });

  it("footer icon loop length", () => {
    expect(wrapper.find(Icon).length).toEqual(5);
  });
  it("checking Icon class should be same", () => {
    expect(
      wrapper
        .find(Icon)
        .find("img")
        .first()
        .props().className
    ).toEqual("iconImage");
  });
  it("checking icon name should be same", () => {
    expect(
      wrapper
        .find(Icon)
        .find("img")
        .first()
        .props().src
    ).toEqual(`${IMAGE_PATH}linkedin.png`);
    expect(
      wrapper
        .find(Icon)
        .find("img")
        .at(1)
        .props().src
    ).toEqual(`${IMAGE_PATH}twitter.png`);
    expect(
      wrapper
        .find(Icon)
        .find("img")
        .at(2)
        .props().src
    ).toEqual(`${IMAGE_PATH}facebook.png`);
    expect(
      wrapper
        .find(Icon)
        .find("img")
        .at(3)
        .props().src
    ).toEqual(`${IMAGE_PATH}youtube.png`);
    expect(
      wrapper
        .find(Icon)
        .find("img")
        .last()
        .props().src
    ).toEqual(`${IMAGE_PATH}insta.png`);
  });
  it("Snapshot test", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
