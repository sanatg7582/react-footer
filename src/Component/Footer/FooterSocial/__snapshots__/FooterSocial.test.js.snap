// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders FooterSocial without crashing  Snapshot test 1`] = `ReactWrapper {}`;
